tools/include/eosio/abigen.hpp

```diff
@@ -530,5 +626,143 @@ namespace eosio { namespace cdt {
          std::set<abi_table>                   ctables;
          std::map<std::string, std::string>    rcs;
          std::set<const clang::Type*>          evaluated;
+
+         bool is_kv_table(const clang::CXXRecordDecl* decl) {
+            for (const auto& base : decl->bases()) {
+               auto type = base.getType();
+               if (type.getAsString().find("eosio::kv_table<") != std::string::npos) {
+                  return true;
+               }
+            }
+
+            return false;
+         }
+
+         std::string get_value_from_decltype(const clang::DecltypeType* decl) {
```

Comment:
This isn't really getting the value from a decltype.  It is getting the std::string representation of the deduced type.